### DISCUSSION TITLE
Long term automation delay

### DISCUSSION THREAD

**Discussion by xlog-null** [Budibase/budibase#8336] [Help] [answered]
Hi, I'm wondering if it's possible to delay an automation for a long time and whether there are any max limits to this duration. I've setup an automation to be delayed for 28 days. The automation can take only milliseconds as far I can tell, so 28 days in milliseconds is 2419200000. I've set it to that, but it has fired immediately, ignoring this duration.

2419200000 is much larger than the standard integer size of 2147483647, so I'm wondering whether that is the issue.

**Comment by melohagan** [comment, marked as answer]
Hey @xlog-null 

I think you're spot on about the standard integer size: https://mrcoles.com/maximum-delay-settimeout/

What might be a better idea is to have a **Cron trigger** that runs once a day. Then you can have an execution date column.
If the date today matches the execution date column, then do your automation stuff, otherwise exit. 
https://docs.budibase.com/docs/conditions-1

**Reply by xlog-null** [reply]
Thank you @melohagan. Does that mean the cron job will run every day even after execution date? Or is there a way to stop the cron after it executed its task?

**Reply by melohagan** [reply]
You could save a completion flag somewhere that could be checked against a condition to prevent the cron being fully executed, but at present there is no way to dynamically delete a cron job. They have to be manually removed.

**Reply by xlog-null** [reply]
Ah I see. Would be nice if I could at least have a cron that runs everyday and pulls all rows (from a MySQL table) that need a task completing and for each row do that task. Thank you for your help :)
